src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonWriterTests.WriteRaw.cs

```diff
@@ -0,0 +1,259 @@
+            {
+                foreach (string str in JsonSerializer.Deserialize<string[]>(data))
+                {
+                    Assert.Equal("Hello", str);
+                }
+            };
+            yield return new object[] { json, validate };
+
+            json = JsonSerializer.SerializeToUtf8Bytes(Enumerable.Repeat("Hello", 4));
+            validate = (data) =>
+            {
+                foreach (string str in JsonSerializer.Deserialize<string[]>(data))
+                {
+                    Assert.Equal("Hello", str);
+                }
+            };
+            yield return new object[] { json, validate };
+        }
+
+        private static byte[] WrapInQuotes(string json)
+        {
+            byte[] buffer = new byte[json.Length + 2];
+            buffer[0] = (byte)'"';
+            Encoding.UTF8.GetBytes(json).CopyTo(buffer, 1);
+            buffer[json.Length + 1] = (byte)'"';
+            return buffer;
+        }
+
+        [Theory]
+        [InlineData(true, 0, "[]")]
+        [InlineData(false, 0, "[]")]
+        [InlineData(true, 1, "[1]")]
+        [InlineData(false, 1, "[1]")]
+        [InlineData(true, 5, "[1,1,1,1,1]")]
+        [InlineData(false, 5, "[1,1,1,1,1]")]
+        public static void WriteRawArrayElements(bool skipInputValidation, int numElements, string expectedJson)
+        {
+            using MemoryStream ms = new();
+            using Utf8JsonWriter writer = new(ms);
+            writer.WriteStartArray();
+
+            for (int i = 0; i < numElements; i++)
+            {
+                writer.WriteRawValue(s_oneAsJson, skipInputValidation);
+            }
+
+            writer.WriteEndArray();
+
+            writer.Flush();
+            Assert.Equal(expectedJson, Encoding.UTF8.GetString(ms.ToArray()));
+        }
+
+        [Theory]
+        [InlineData(true, 0, "{}")]
+        [InlineData(false, 0, "{}")]
+        [InlineData(true, 1, @"{""int"":1}")]
+        [InlineData(false, 1, @"{""int"":1}")]
+        [InlineData(true, 3, @"{""int"":1,""int"":1,""int"":1}")]
+        [InlineData(false, 3, @"{""int"":1,""int"":1,""int"":1}")]
+        public static void WriteRawObjectProperty(bool skipInputValidation, int numElements, string expectedJson)
+        {
+            using MemoryStream ms = new();
+            using Utf8JsonWriter writer = new(ms);
+            writer.WriteStartObject();
+
+            for (int i = 0; i < numElements; i++)
+            {
+                writer.WritePropertyName("int");
+                writer.WriteRawValue(s_oneAsJson, skipInputValidation);
+            }
+
+            writer.WriteEndObject();
+
+            writer.Flush();
+            Assert.Equal(expectedJson, Encoding.UTF8.GetString(ms.ToArray()));
+        }
+
+        [Theory]
+        [InlineData("[")]
+        [InlineData("}")]
+        [InlineData("[}")]
+        [InlineData("xxx")]
+        [InlineData("{hello:")]
+        public static void WriteRawInvalidJson(string json)
```

Comment:
We should expand on the invalid json writing test cases, and include invalid encoded data.

We also should add tests for writing valid json, but that is structurally invalid given the state of the writer.